sidebar/src/components/deploy/contract/DeployUsingContract.logic.ts

```diff
@@ -26,25 +27,26 @@ export const useDeployUsingContract = (
       switch (event.data.type) {
         case MessageType.DEPLOY_CONTRACT_RESPONSE: {
           setResponse(event.data.response);
+          setIsPending(false);
           break;
         }
       }
     };
     window.addEventListener('message', listener);
     return () => window.removeEventListener('message', listener);
-  }, []);
+  }, [setIsPending]);
 
   useEffect(() => {
     form.setValue('wallet', wallets[0]?.id || '');
-  }, [wallets]);
+  }, [wallets, form]);
```

Comment:
Avoid using form in dependencies, otherwise the wallet will be set everytime form change and we don't want that